### ISSUE TITLE
Unexpected result when comparing differing types

### ISSUE THREAD

**Issue by faunaee** [netilon/differify#11]
Example:

```js
const Differify = require('@netilon/differify')
const util = require('util')

const a = [ 1, 2, 3 ]
const b = { a: "foo", b: "bar" }

var differify = new Differify({
  mode: { object: 'DIFF', array: 'DIFF' }
})
var diff = differify.compare(a, b)
console.log(util.inspect(diff, { depth: null }))
```

Output:

```
{
  original: [ 1, 2, 3 ],
  current: { a: 'foo', b: 'bar' },
  status: 'MODIFIED',
  changes: 0
}
```

Since these are differing types, and the status is `MODIFIED`, I would have expected the `changes` field to report a non-zero value. I can see that it makes a certain kind of sense that `value` is zero in this situation. But taking these fields together tells me "there's a diff, but there are no differences", which is harder for dependent code (e.g. a diff visualizer) to reason about.

Perhaps a new status of `CANNOT_COMPARE` should be added?

If nothing else, an example in the documentation would be helpful.

**Comment by fabian-orue** [comment]
Hi @faunaee, thank you for letting me know,  i will check it.

**Comment by fabian-orue** [comment]
Hi @faunaee! the fix to this issue is now available in the new released version **3.2.1**. Regards

**Comment by faunaee** [comment]
Great! Thanks very much!
